cwalk.go

```diff
@@ -19,11 +19,27 @@ var BufferSize = NumWorkers
 // to a walker function, does not point to a directory
 var ErrNotDir = errors.New("Not a directory")
 
+// A struct to store the errors reported from each worker routine
+type WalkerError struct {
+	ErrorList []error
+}
+
+// Implement the error interface
+func (we WalkerError) Error() string {
+	if len(we.ErrorList) > 0 {
+		return "There were errors walking files"
```

Comment:
I think that it would be better for the Error() method to actually return a rendered list of errors, i.e. something like that:

```
		out := make([]string, len(we.ErrorList))
		for i, err := range we.ErrorList {
                         out[i] = err.Error()
		}
                return strings.Join(out, "\n")
```

Reply:
OK